### DISCUSSION TITLE
How to include data files in generated macOS .app bundle

### DISCUSSION THREAD

**Discussion by zevlee** [pyinstaller/pyinstaller#6733] [Help] [answered]
Is it possible to include data files when generating a macOS .app bundle? Adding the ``app = BUNDLE()`` option to my spec file forces pyinstaller to work with the ``--onefile`` flag, which results in the extra files not being included.

In my current build script for [Passphraser](https://github.com/zevlee/passphraser/blob/main/macos/build.sh#L10-L22), I have to manually create the .app bundle after generating the directory with pyinstaller. I'd prefer to have pyinstaller generate the bundle outright, if possible.

Here's the spec file I'm trying to use:

```
# -*- mode: python ; coding: utf-8 -*-

block_cipher = None

a = Analysis(
	['../passphraser'],
	pathex=[],
	binaries=[],
	datas=[
		('../lib/__init__.py', 'lib'),
		('../lib/passphraser.py', 'lib'),
		('../gui/menu.xml', 'gui'),
		('../passphraser.svg', '.'),
		('../wordlists/eff_large.txt', 'wordlists'),
		('../wordlists/eff_short_1.txt', 'wordlists'),
		('../wordlists/eff_short_2.txt', 'wordlists'),
		('../LICENSE', '.'),
		('../VERSION', '.')
	],
	hiddenimports=[],
	hookspath=[],
	runtime_hooks=[],
	excludes=[],
	win_no_prefer_redirects=False,
	win_private_assemblies=False,
	cipher=block_cipher,
	noarchive=False
)

pyz = PYZ(
	a.pure,
	a.zipped_data,
	cipher=block_cipher
)

exe = EXE(
	pyz,
	a.scripts,
	[],
	exclude_binaries=True,
	name='passphraser',
	icon='passphraser.icns',
	debug=False,
	bootloader_ignore_signals=False,
	strip=False,
	upx=True,
	console=False
)

coll = COLLECT(
	exe,
	a.binaries,
	a.zipfiles,
	a.datas,
	strip=False,
	upx=True,
	upx_exclude=[],
	name='passphraser'
)

app = BUNDLE(
	exe,
	name='Passphraser.app',
	icon='passphraser.icns',
	bundle_identifier='me.zevlee.Passphraser',
	version='VERSION'
)
```

**Comment by rokm** [comment, marked as answer]
> Adding the `app = BUNDLE()` option to my spec file forces pyinstaller to work with the `--onefile` flag, which results in the extra files not being included.

That's because you're passing an incorrect argument (`exe`) to the `BUNDLE`, and so only the generated executable gets collected into .app bundle. 

For `onedir` builds, `coll` should be passed to `BUNDLE` instead:
https://github.com/pyinstaller/pyinstaller/blob/3940e5fc5f9e02bce9f1af53c70a42db81071381/PyInstaller/building/templates.py#L125-L131

**Reply by zevlee** [reply]
Ah yes, that was the issue. Thank you very much for pointing my error out. The bundle works now.
